### DISCUSSION TITLE
[Android] update to v10.0.1 issue

### DISCUSSION THREAD

**Discussion by moiatto** [maplibre/maplibre-native#849] [Q&A]
Short story:

The problem start from the last line:
```
MapboxNavigationOptions options = MapboxNavigationOptions.builder()
                .isDebugLoggingEnabled(true)
                .build();
navigation = new MapboxNavigation(requireContext(), options);
navigation.addNavigationEventListener(this);
navigation.addMilestoneEventListener(this);
navigation.setLocationEngine(LocationEngineProvider.getBestLocationEngine(requireContext()));
```

I have this dependencies in my app
```
implementation 'com.google.android.gms:play-services-location:21.0.1'
implementation 'com.github.maplibre:maplibre-navigation-android:84e9f77a55' 
implementation 'org.maplibre.gl:android-sdk:9.6.0'    
```
The gms v21.0.1 is not compatible with maplibre-gl v9.6.0, throwing this error:
```
java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient....
```
I have tried to update maplibre-gl to the last version v10.0.1, but there are 2 problems;

1. I don't understand the documentation to obtain a valid LocationEngine to pass in navigation.setLocationEngine
2. On run app I get errors about icons not found linked in the library file values.xml at pos: 9711:
```
....
    <style name="mapbox_LocationLayer">
        <item name="mapbox_foregroundDrawable">@drawable/mapbox_user_icon</item>
        <item name="mapbox_backgroundDrawable">@drawable/mapbox_user_stroke_icon</item>
        <item name="mapbox_bearingDrawable">@drawable/mapbox_user_bearing_icon</item>
        <item name="mapbox_gpsDrawable">@drawable/mapbox_user_puck_icon</item>

        <item name="mapbox_foregroundDrawableStale">@drawable/mapbox_user_icon_stale</item>
        <item name="mapbox_backgroundDrawableStale">@drawable/mapbox_user_stroke_icon</item>

        <item name="mapbox_accuracyAlpha">0.15</item>
        <item name="mapbox_accuracyColor">@color/mapbox_location_layer_blue</item>
....
```
where can i find these icons?

So, the alternative is to use Android GPS and Network Providers but 
how can I force LocationEngineProvider to use Android GPS and Network Providers instead Google's Fused Location Providers ?

**Comment by louwers** [comment]
MapboxNavigationOptions is not a class in MapLibre GL Native for Android. Try one of these repos instead to get support for it:

- https://github.com/mapbox/mapbox-navigation-android
- https://github.com/maplibre/maplibre-navigation-android

`LocationEngineProvider` has been removed.

You can create a `LocationEngine` as follows:

```kotlin
new LocationEngineProxy<>(new MapboxFusedLocationEngineImpl(context.getApplicationContext()))
```

**Reply by moiatto** [reply]
I tried this, but I receive this error:
```
error: MapboxFusedLocationEngineImpl(Context) is not public in MapboxFusedLocationEngineImpl; cannot be accessed from outside package
LocationEngine locationEngine = new LocationEngineProxy(new MapboxFusedLocationEngineImpl(context.getApplicationContext()));
```
                                                                ^

**Reply by louwers** [reply]
That will be fixed with the 10.0.2 release. Please hang tight. :slightly_smiling_face:

**Reply by moiatto** [reply]
Thank you. you have been very kind
